shell/platform/fuchsia/dart-pkg/zircon/lib/src/system.dart

```diff
@@ -0,0 +1,161 @@
+      return 'ZxStatusException: status = $status';
+    else
+      return 'ZxStatusException: status = $status, "$message"';
+  }
+}
+
+class _Result {
+  final int status;
+  const _Result(this.status);
+}
+
+@pragma('vm:entry-point')
+class HandleResult extends _Result {
+  final Handle handle;
+  @pragma('vm:entry-point')
+  const HandleResult(final int status, [this.handle]) : super(status);
+  @override
+  String toString() => 'HandleResult(status=$status, handle=$handle)';
+}
+
+@pragma('vm:entry-point')
+class HandlePairResult extends _Result {
+  final Handle first;
+  final Handle second;
+  @pragma('vm:entry-point')
+  const HandlePairResult(final int status, [this.first, this.second])
+      : super(status);
+  @override
+  String toString() =>
+      'HandlePairResult(status=$status, first=$first, second=$second)';
+}
+
+@pragma('vm:entry-point')
+class ReadResult extends _Result {
+  final ByteData bytes;
+  final int numBytes;
+  final List<Handle> handles;
+  @pragma('vm:entry-point')
+  const ReadResult(final int status, [this.bytes, this.numBytes, this.handles])
+      : super(status);
+  Uint8List bytesAsUint8List() =>
+      bytes.buffer.asUint8List(bytes.offsetInBytes, numBytes);
+  String bytesAsUTF8String() => utf8.decode(bytesAsUint8List());
+  @override
+  String toString() =>
+      'ReadResult(status=$status, bytes=$bytes, numBytes=$numBytes, handles=$handles)';
+}
+
+@pragma('vm:entry-point')
+class WriteResult extends _Result {
+  final int numBytes;
+  @pragma('vm:entry-point')
+  const WriteResult(final int status, [this.numBytes]) : super(status);
+  @override
+  String toString() => 'WriteResult(status=$status, numBytes=$numBytes)';
+}
+
+@pragma('vm:entry-point')
+class GetSizeResult extends _Result {
+  final int size;
+  @pragma('vm:entry-point')
+  const GetSizeResult(final int status, [this.size]) : super(status);
+  @override
+  String toString() => 'GetSizeResult(status=$status, size=$size)';
+}
+
+@pragma('vm:entry-point')
+class FromFileResult extends _Result {
+  final Handle handle;
+  final int numBytes;
+  @pragma('vm:entry-point')
+  const FromFileResult(final int status, [this.handle, this.numBytes])
+      : super(status);
+  @override
+  String toString() =>
+      'FromFileResult(status=$status, handle=$handle, numBytes=$numBytes)';
+}
+
+@pragma('vm:entry-point')
+class MapResult extends _Result {
+  final Uint8List data;
+  @pragma('vm:entry-point')
+  const MapResult(final int status, [this.data]) : super(status);
+  @override
+  String toString() => 'MapResult(status=$status, data=$data)';
+}
+
+@pragma('vm:entry-point')
+class System extends NativeFieldWrapperClass2 {
+  // No public constructor - this only has static methods.
+  System._();
+
+  // Channel operations.
+  static HandlePairResult channelCreate([int options = 0])
+      native 'System_ChannelCreate';
+  static HandleResult channelFromFile(String path)
+      native 'System_ChannelFromFile';
+  static int reboot()
```

Comment:
Are you planning to include updates still happening in the Fuchsia tree like https://fuchsia-review.googlesource.com/c/topaz/+/275857 ?

Reply:
Yes. I am keeping a list of these patches in the "Patches to Backport" section of go/fdk-notes.